tests/SimplifyTest.elm

```diff
@@ -14976,6 +14977,174 @@ a = False
+                            , under = "Dict.partition"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a = ( Dict.empty, Dict.empty )
+"""
+                        ]
+        , test "should replace Dict.empty |> Dict.partition f by ( Dict.empty, Dict.empty )" <|
+            \() ->
+                """module A exposing (..)
+a = Dict.empty |> Dict.partition f
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "Using Dict.partition on Dict.empty will result in ( Dict.empty, Dict.empty )"
+                            , details = [ "You can replace this call by ( Dict.empty, Dict.empty )." ]
+                            , under = "Dict.partition"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a = ( Dict.empty, Dict.empty )
+"""
+                        ]
+        , test "should replace Dict.partition (always True) x by ( x, Dict.empty )" <|
+            \() ->
+                """module A exposing (..)
+a = Dict.partition (always True) x
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "All elements will go to the first Dict"
+                            , details = [ "Since the predicate function always returns True, the second Dict will always be Dict.empty." ]
+                            , under = "Dict.partition"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a = ( x, Dict.empty )
+"""
+                        ]
+        , test "should replace Dict.partition (\\_ -> True) x by ( x, Dict.empty )" <|
+            \() ->
+                """module A exposing (..)
+a = Dict.partition (\\_ -> True) x
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "All elements will go to the first Dict"
+                            , details = [ "Since the predicate function always returns True, the second Dict will always be Dict.empty." ]
+                            , under = "Dict.partition"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a = ( x, Dict.empty )
+"""
+                        ]
+        , test "should not replace Dict.partition (always True)" <|
+            -- We'd likely need an anonymous function which could introduce naming conflicts
+            -- Could be improved if we knew what names are available at this point in scope (or are used anywhere)
+            -- so that we can generate a unique variable.
```

Comment:
We now have the means to create argument names that aren't locally shadowed. I don't really like the lambda fix, tho

Reply:
Oh yeah, we could do that now :eyes: 
The naming convention we choose could displease people, but as a suggestion, it could definitely be a good start (and that's partially why there are prompts for fixes).